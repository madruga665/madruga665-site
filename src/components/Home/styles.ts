import styled from "styled-components";
import { AnimationPositionLeft, AnimationPositionRight } from "../../styles/animations";
import { Colors } from "../../styles/variables";

export const AboutContainer = styled("section")`
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1.5rem 1rem;
  margin-top: 8rem;

  @media only screen and (min-width: 768px) {
    width: 90%;
    flex-direction: row;
    justify-content: center;
    margin: 0;
  }
`;

export const ImageContainer = styled("div")`
  margin: 0 0 1rem 0;
  animation: ${AnimationPositionLeft} 2.5s ease-in-out;

  .about-image {
    border: 5px solid ${Colors.emerald};
    border-radius: 50%;
    width: 350px;
  }
`;

export const TextContainer = styled("div")`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0.5rem;
  animation: ${AnimationPositionRight} 2.5s ease-in-out;
  max-width: 30rem;

  & p {
    font-size: 1.5rem;
    text-align: center;
    color: #fff;

    @media only screen and (min-width: 768px) {
      margin-left: 1rem;
      text-align: left;
    }
  }
`;
