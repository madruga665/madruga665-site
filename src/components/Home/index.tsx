import { useEffect, useState } from "react";
import { Puff } from "react-loader-spinner";
import axios from 'axios';

import styles from "./styles.module.scss";
import colors from "../../styles/colors.module.scss";

function Home() {
  const [presentation, setPresentation] = useState("");

  const teste = async () => {
    const response = await axios.get('https://madruga665.vercel.app/api/notion');
    const data = await response.data;

    setPresentation(data);
  };

  useEffect(() => {
    teste();
  }, []);

  return (
    <div className={styles.AboutContainer}>
      {!presentation ? (
        <Puff
          height='140'
          width='140'
          radius={1}
          color={colors.colorEmerald}
          ariaLabel='puff-loading'
          wrapperStyle={{}}
          wrapperClass=''
          visible={true}
        />
      ) : (
        <>
          <div className={styles.ImageContainer}>
            <img
              className={styles["about-image"]}
              src='https://github.com/madruga665.png'
              alt='Luciano Amancio'
            />
          </div>
          <div className={styles.TextContainer}>
            <p>{presentation}</p>
          </div>
        </>
      )}
    </div>
  );
}

export default Home;
