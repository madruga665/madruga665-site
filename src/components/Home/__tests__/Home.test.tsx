import { render, screen } from "@testing-library/react";
import Home from "..";

describe("Testes do componente About", () => {
  test("Snapshot do componente", () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  test("A imagem esta sendo renderizado", () => {
    render(<Home />);
    const madrugaImage = screen.getByRole("img", {
      name: /luciano amancio/i,
    });

    expect(madrugaImage).toBeInTheDocument();
  });

  test("O texto 'sobre' está sendo renderizado", () => {
    render(<Home />);
    const aboutText = screen.getByText(
      /olá, me chamo luciano amâncio, tenho 36 anos e moro em guarujá\-sp\. sou pai da julia do heitor e do dante\. sou desenvolvedor web full stack, formado pela trybe e atualmente trabalho na remessa online\. criei esse site para mostrar um pouco de minhas habilidades e experiências\./i
      )

    expect(aboutText).toBeInTheDocument();
  });
});
