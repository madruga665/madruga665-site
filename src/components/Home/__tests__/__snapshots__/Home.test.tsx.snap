// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do componente About Snapshot do componente 1`] = `
<div>
  <section
    class="sc-bdvvtL nNfhK"
    id="sobre"
  >
    <div
      class="sc-gsDKAQ dcfYXI"
    >
      <img
        alt="Luciano Amancio"
        class="about-image"
        src="https://github.com/madruga665.png"
      />
    </div>
    <div
      class="sc-dkPtRN gkuLoe"
    >
      <p>
        Olá, me chamo Luciano Amâncio, tenho 36  anos e moro em Guarujá-SP. Sou pai da Julia do
  Heitor e do Dante. Sou Desenvolvedor Web Full Stack, formado pela Trybe e atualmente
  trabalho na Remessa Online. Criei esse site para mostrar um pouco de minhas habilidades e
  experiências.
      </p>
    </div>
  </section>
</div>
`;
